src/framework/ui/internal/platform/macos/macosplatformtheme.mm

```diff
@@ -18,51 +18,103 @@
 //=============================================================================
 
 #include "macosplatformtheme.h"
+#include "uitypes.h"
+#include "log.h"
 
 #include <Cocoa/Cocoa.h>
+#include <QWidget>
 
 using namespace mu::ui;
 using namespace mu::async;
 
-id<NSObject> darkModeObserverToken;
+id<NSObject> darkModeObserverToken = nil;
+id<NSObject> contrastObserverToken = nil;
 
 void MacOSPlatformTheme::startListening()
 {
-    NSDistributedNotificationCenter* n = [NSDistributedNotificationCenter defaultCenter];
-    darkModeObserverToken = [n addObserverForName:@"AppleInterfaceThemeChangedNotification"
-                                           object:nil
-                                            queue:nil
-                                       usingBlock:^(NSNotification*) { m_darkModeSwitched.send(isDarkMode()); }];
+    if (!darkModeObserverToken) {
+        darkModeObserverToken = [[NSDistributedNotificationCenter defaultCenter]
+                                 addObserverForName:@"AppleInterfaceThemeChangedNotification"
+                                 object:nil
+                                 queue:nil
+                                 usingBlock:^(NSNotification*) {
+            m_channel.send(bestSuitedThemeCode());
+        }];
+    }
+
+    if (!contrastObserverToken) {
+        contrastObserverToken = [[[NSWorkspace sharedWorkspace] notificationCenter]
+                                 addObserverForName:NSWorkspaceAccessibilityDisplayOptionsDidChangeNotification
+                                 object:nil
+                                 queue:nil
+                                 usingBlock:^(NSNotification*) {
+            m_channel.send(bestSuitedThemeCode());
+        }];
+    }
 }
 
 void MacOSPlatformTheme::stopListening()
 {
-    NSDistributedNotificationCenter* n = [NSDistributedNotificationCenter defaultCenter];
-    [n removeObserver:darkModeObserverToken];
+    if (darkModeObserverToken) {
+        [[NSDistributedNotificationCenter defaultCenter] removeObserver:darkModeObserverToken];
+        darkModeObserverToken = nil;
+    }
+
+    if (contrastObserverToken) {
+        [[[NSWorkspace sharedWorkspace] notificationCenter] removeObserver:contrastObserverToken];
+        contrastObserverToken = nil;
+    }
 }
 
 bool MacOSPlatformTheme::isFollowSystemThemeAvailable() const
 {
+    // Supported from macOS 10.14, which is our minimum supported version
     return true;
 }
 
-bool MacOSPlatformTheme::isDarkMode() const
+std::string MacOSPlatformTheme::bestSuitedThemeCode() const
+{
+    if (isSystemDarkMode()) {
+        if (isSystemHighContrast()) {
+            return HIGH_CONTRAST_THEME_CODE;
+        } else {
+            return DARK_THEME_CODE;
+        }
+    } else {
```

Comment:
You can remove this else and just use return:

if (isSystemDarkMode()) {
...
}

return LIGHT_THEME_CODE;